src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheckTest.java

```diff
@@ -269,6 +272,73 @@ public void testVisitLeaveToken()
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(NonTightHtmlTagIntolerantCheck.class);
+        checkConfig.addAttribute("violateExecutionOnNonTightHtml", "true");
+        final String[] expected = {
+            "6: " + getCheckMessage(MSG_UNCLOSED_HTML, "p"),
+            "13: " + getCheckMessage(MSG_UNCLOSED_HTML, "p"),
+            "16: " + getCheckMessage(MSG_UNCLOSED_HTML, "li"),
+            "21: " + getCheckMessage(MSG_UNCLOSED_HTML, "p"),
+            "27: " + getCheckMessage(MSG_UNCLOSED_HTML, "tr"),
+            "34: " + getCheckMessage(MSG_UNCLOSED_HTML, "li"),
+            "40: " + getCheckMessage(MSG_UNCLOSED_HTML, "th"),
+            "54: " + getCheckMessage(MSG_UNCLOSED_HTML, "li"),
+            "71: " + getCheckMessage(MSG_UNCLOSED_HTML, "p"),
+            "76: " + getCheckMessage(MSG_UNCLOSED_HTML, "p"),
+            "85: " + getCheckMessage(MSG_UNCLOSED_HTML, "td"),
+            "90: " + getCheckMessage(MSG_UNCLOSED_HTML, "html"),
+            "97: " + getCheckMessage(MSG_UNCLOSED_HTML, "thead"),
+            "124: " + getCheckMessage(MSG_UNCLOSED_HTML, "p"),
+        };
+        verify(checkConfig, getPath("InputAbstractJavadocNonTightHtmlTags.java"), expected);
+    }
+
+    @Test
+    public void testNonTightHtmlTagIntolerantCheckReportingNoViolation() throws Exception {
+        NonTightHtmlTagIntolerantCheck.visitJavadocTokenMethodCallCount = 0;
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(NonTightHtmlTagIntolerantCheck.class);
+        final String[] expected = CommonUtils.EMPTY_STRING_ARRAY;
+        verify(checkConfig, getPath("InputAbstractJavadocNonTightHtmlTags.java"), expected);
+    }
+
+    @Test
+    public void testNonTightHtmlTagIntolerantCheckVisitJavadocTokenMethodCallCount()
+            throws Exception {
+        NonTightHtmlTagIntolerantCheck.visitJavadocTokenMethodCallCount = 0;
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(NonTightHtmlTagIntolerantCheck.class);
+        final File testFile = new File(getPath("InputAbstractJavadocNonTightHtmlTags.java"));
+        final List<File> fileList = new ArrayList<>();
+        fileList.add(testFile);
+        final Checker checker = createChecker(checkConfig);
+        checker.process(fileList);
+        Assert.assertEquals("Please ensure that method 'visitJavadocToken' is not called "
+                        + "more than thrice.",
+                3, NonTightHtmlTagIntolerantCheck.visitJavadocTokenMethodCallCount);
+    }
+
+    @Test
+    public void testVisitCountForCheckAcceptingJavadocWithNonTightHtml()
+            throws Exception {
+        NonTightHtmlTagTolerantCheck.visitJavadocTokenMethodCallCount = 0;
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(NonTightHtmlTagTolerantCheck.class);
+        checkConfig.addAttribute("violateExecutionOnNonTightHtml", "true");
```

Comment:
Please ensure if `violateExecutionOnNonTightHtml` and `doesCheckAcceptJavadocWithNonTightHtml()` can vary independently and `violateExecutionOnNonTightHtml = true;` and `doesCheckAcceptJavadocWithNonTightHtml()` returning true for a check is a valid combination